src/components/form/templates/array-field-collapsible-item.ts

```diff
@@ -38,6 +38,8 @@ export class CollapsibleItemTemplate extends React.Component {
     constructor(public props: CollapsibleItemProps) {
         super(props);
         this.handleAction = this.handleAction.bind(this);
+        this.isDeepEmpty = this.isDeepEmpty.bind(this);
```

Comment:
This accomplishes the same thing, and doesn't run the same risk of someone (probably future me) accidentally reverting the change in an attempt to "clean up" 😄

Reply:
seriously? :laughing: I thought about that. but then why is the recursive call working as long as it's not array of objects?

Reply:
I would assume because we're in `array-field-collapsible-item.ts`, so this code isn't used unless we're working with an array? Not sure though, I haven't investigated any further.

(And yes, my first reaction when I saw your change was "WAT? How is this changing anything?" — which is also precisely why I foresee my future self coming across that and trying to "clean it up", thereby breaking it again 😂)

Reply:
not really, `array-field-collapsible-item` is always only one item (the array gets handle in `array-field`). for section though we have an array of controls (so a collapsible in a collapsible) and even that works as long as `isDeepEmpty` finds a value before reaching the controls array. but when you add a new section it has to check all of it and that's when it failed

Reply:
I have no idea then 😄

Good find either way! 👏👏